tests/unit/admin/test-class-wc-rest-payments-orders-controller.php

```diff
@@ -534,6 +534,399 @@ public function test_capture_terminal_payment_not_found() {
 		$this->assertEquals( 404, $data['status'] );
 	}
 
+	public function test_capture_authorization_success() {
+		$order = $this->create_mock_order();
+		$order->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
+		$order->set_payment_method_title( 'WooCommerce Payments' );
+		$order->save();
+
+		$mock_intent = WC_Helper_Intention::create_intention(
+			[
+				'status'   => 'requires_capture',
+				'metadata' => [
+					'order_id' => $order->get_id(),
+				],
+			]
+		);
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_intent' )
+			->willReturn( $mock_intent );
+
+		$this->mock_gateway
+			->expects( $this->once() )
+			->method( 'capture_charge' )
+			->with( $this->isInstanceOf( WC_Order::class ) )
+			->willReturn(
+				[
+					'status' => 'succeeded',
+					'id'     => $this->mock_intent_id,
+				]
+			);
+
+		$this->mock_gateway
+			->expects( $this->never() )
+			->method( 'attach_intent_info_to_order' );
+
+		$this->mock_gateway
+			->expects( $this->once() )
+			->method( 'update_order_status_from_intent' )
+			->with(
+				$this->isInstanceOf( WC_Order::class ),
+				$this->mock_intent_id,
+				'requires_capture',
+				'ch_mock'
+			);
+
+		$request = new WP_REST_Request( 'POST' );
+		$request->set_body_params(
+			[
+				'order_id'          => $order->get_id(),
+				'payment_intent_id' => $this->mock_intent_id,
+			]
+		);
+
+		$response      = $this->controller->capture_authorization( $request );
+		$response_data = $response->get_data();
+
+		$this->assertEquals( 200, $response->status );
+		$this->assertEquals(
+			[
+				'status' => 'succeeded',
+				'id'     => $this->mock_intent_id,
+			],
+			$response_data
+		);
+
+		$result_order = wc_get_order( $order->get_id() );
+		$this->assertEquals( 'woocommerce_payments', $result_order->get_payment_method() );
+		$this->assertEquals( 'WooCommerce Payments', $result_order->get_payment_method_title() );
+		$this->assertEquals( 'processing', $result_order->get_status() );
+		$url = '/wc/v3/' . ( $this->is_wpcom() ? 'sites/3/' : '' ) . 'payments/readers/receipts/' . $this->mock_intent_id;
+		$this->assertStringEndsWith( $url, $result_order->get_meta( 'receipt_url' ) );
```

Comment:
Is it necessary to assert reader receipt here?

Reply:
You are right, it is not needed since the receipt_url is intended for Terminal payments. Removed in [85d34d2](https://github.com/Automattic/woocommerce-payments/pull/4676/commits/85d34d2bf9646943845321a37e726feb0182b1d9)